### PR TITLE
ci: automate releases on merge of release/ios-sdk-v* PRs

### PR DESCRIPTION
**Pull request by wangerekaharun**
### **User description**
Story: https://app.shortcut.com/smileid/story/xxx

## Summary

A few sentences/bullet points about the changes

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

___

### **PR Type**
Enhancement

___

### **Description**
- Add automated release workflow on merge of `release/ios-sdk-v*` PRs

- Remove legacy `post_release` and `sdk-release` workflows

- Tag merge commit, create GitHub release with ios-sdk artifacts

- Push podspecs to CocoaPods Trunk and sync CHANGELOG via PR

___

### Diagram Walkthrough

```mermaid
flowchart LR
  A["PR merged to main"] -- "branch matches release/ios-sdk-v*" --> B["Extract version"]
  B --> C["Tag merge commit"]
  C --> D["Download ios-sdk artifacts"]
  D --> E["Create GitHub Release"]
  E --> F["Push podspecs to CocoaPods"]
  F --> G["Sync CHANGELOG & open PR"]
```

<details> <summary><h3> File Walkthrough</h3></summary>

<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>release-on-merge.yml</strong><dd><code>Add automated release-on-merge workflow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/release-on-merge.yml

<ul><li>New workflow triggered when <code>release/ios-sdk-v*</code> PRs are merged to <code>main</code><br> <li> Extracts version from branch name, tags the merge commit, and <br>force-pushes the tag<br> <li> Downloads release artifacts (<code>.zip</code>, <code>.sha256</code>) from the private <code>ios-sdk</code> <br>repo and creates a public GitHub release<br> <li> Installs CocoaPods, pushes <code>SmileIDSDK.podspec</code> then <code>SmileID.podspec</code> to <br>Trunk with a 5-minute wait between them<br> <li> Fetches the matching CHANGELOG section from <code>ios-sdk</code>, prepends it to <br>the local <code>CHANGELOG.md</code>, and opens a PR for review</ul>

</details>

  </td>
  <td><a href="https://github.com/smileidentity/ios/pull/476/files#diff-014acb62cbe0b3a074e82cc7d47b4c76a9374b1846d450780f36a3e7432192fd">+158/-0</a>&nbsp; </td>

</tr>
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>post_release.yml</strong><dd><code>Remove legacy post-release workflow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/post_release.yml

<ul><li>Removed the entire <code>Post-Release Actions</code> workflow that previously <br>triggered on GitHub release creation to sync CHANGELOG to the docs <br>repo</ul>

</details>

  </td>
  <td><a href="https://github.com/smileidentity/ios/pull/476/files#diff-7235ed47d43e54136a9c269dc833490c0d3b86f527f83edb114c879fa4ba6f66">+0/-34</a>&nbsp; &nbsp; </td>

</tr>

<tr>
  <td>
    <details>
      <summary><strong>sdk-release.yml</strong><dd><code>Remove manual sdk-release workflow</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

.github/workflows/sdk-release.yml

<ul><li>Removed the manual <code>Release iOS SDK</code> workflow_dispatch workflow that <br>delegated to <code>sdk-primary.yml</code></ul>

</details>

  </td>
  <td><a href="https://github.com/smileidentity/ios/pull/476/files#diff-704dfa813b791458c12d01ae0986baeed20c04c8c464932d90e1f9502239786c">+0/-22</a>&nbsp; &nbsp; </td>

</tr>
</table></td></tr></tr></tbody></table>

</details>

___

___

> <details> <summary>  Need help?</summary><li>Type <code>/help how to ...</code> in the comments thread for any questions about PR-Agent usage.</li><li>Check out the <a href="https://qodo-merge-docs.qodo.ai/usage-guide/">documentation</a> for more information.</li></details>

### CODE REVIEW

**Review by jumaallan — DISMISSED** [review]

File: .github/workflows/release-on-merge.yml [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+          name: SmileID iOS SDK v${{ env.RELEASE_VERSION }}
+          target_commitish: ${{ env.MERGE_SHA }}
+          make_latest: true
+          generate_release_notes: true
+          fail_on_unmatched_files: true
+          files: |
+            artifacts/*.zip
+            artifacts/*.zip.sha256
+
+      - name: Setup Ruby and CocoaPods
```
Comment (Yeboahmedia) [inline .github/workflows/release-on-merge.yml:131]: are we supporting cocoapods? @wangerekaharun

Comment (wangerekaharun) [reply .github/workflows/release-on-merge.yml:131]: This is for v11; we still publish the releases to Cocoapods. The wrappers are also not migrated to SPM, so they still use the Cocoapods releases

**Review by jumaallan — APPROVED** [review]
